torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -131,6 +130,28 @@ def _assert_tensors_nonaliasing(inputs, outputs):
     ), "inputs to function body cannot alias outputs"
 
 
+def _check_supported_callable_arg(func_var: VariableTracker, arg_name):
+    from . import (
+        NestedUserFunctionVariable,
+        NNModuleVariable,
+        TorchInGraphFunctionVariable,
+        UnspecializedNNModuleVariable,
+        UserFunctionVariable,
+    )
+
+    if not isinstance(
+        func_var,
+        (
+            UserFunctionVariable,
+            NestedUserFunctionVariable,
+            NNModuleVariable,
+            UnspecializedNNModuleVariable,
+            TorchInGraphFunctionVariable,
+        ),
+    ):
+        unimplemented(f"{arg_name} is of unsupported callable type {str(func_var)}.")
```

Comment:
Is it possible to just run `callable(func_var)` ?

Reply:
That's a good idea but these dynamo variables don't have a `__call__` method. It iis handled as call_function ("__call__"...). We may test callable for the wrapped actual python object though but it seems requiring a reliable way of getting the inner python object out. We can revisit it in the future.

Reply:
torch.compile seems like it supports checking if an object `callable`, so we could figure out how to leverage that

<img width="608" alt="image" src="https://github.com/pytorch/pytorch/assets/5652049/99e55a4a-d332-4835-9432-25f977d43c06">

Reply:
Oh, yeah, we could do similar things like how pytree.flatten in dynamo. I can change to that.